src/copyist/cli.py

```diff
@@ -67,13 +73,20 @@ def main():
         sys.exit(1)
 
     try:
-        sync.sync_files(
-            file_generators, context, verbose=options.verbose, dry_run=options.dry_run,
+        changed = sync.sync_files(
+            file_generators,
+            context,
+            verbose=options.verbose,
+            dry_run=options.dry_run or options.check,
         )
     except exceptions.SyncException as e:
         print(e.args[0])
         sys.exit(1)
 
+    if changed and options.check:
+        print(f"Unsynced files: {', '.join(changed)}")
+        sys.exit(1)
```

Comment:
We could distinguish errors from unsynchronized files by exiting with a different error code.

Reply:
Done in f52396898f